stock_reception_screen/models/stock_reception_screen.py

```diff
@@ -0,0 +1,528 @@
+            package = self.current_move_line_id.result_package_id
+            package.product_packaging_id = packaging
+            # Set the storage type on the package
+            storage_type = packaging.package_storage_type_id
+            package.package_storage_type_id = storage_type
+            # Set the height on the package
+            package.height = packaging.height
+
+    def process_select_product(self):
+        self.next_step()
+        if self.current_move_id:
+            # Go to the next step automatically if only one move has been found
+            self.process_select_move()
+
+    def _validate_current_move(self):
+        """Split the current move with the move line qty done and
+        validate it.
+        It is performed right after the processing of the current move (so
+        before checking the next move to process).
+        """
+        if self.current_move_line_id and self.current_move_id:
+            # We use the 'is_scrap' context key to avoid the generation of a
+            # backorder when validating the move (see _action_done() method in
+            # stock/models/stock_move.py).
+            self.current_move_id.with_context(is_scrap=True)._action_done()
+            if self.picking_id.state == "confirmed":
+                self.picking_id.action_assign()
+
+    def _before_set_location_to_select_move(self):
+        """Check if there is remaining moves to process for the
+        selected product.
+        """
+        self._validate_current_move()
+        if not self.current_filter_product:
+            return False
+        moves_to_process_ok = any(
+            move.quantity_done < move.product_uom_qty
+            for move in self.picking_filtered_move_lines
+        )
+        if moves_to_process_ok:
+            self.current_move_id = False
+            self.current_move_line_id = False
+        return moves_to_process_ok
+
+    def _before_set_location_to_select_product(self):
+        """Check if there is remaining products/moves to process."""
+        moves_to_process_ok = any(
+            move.quantity_done < move.product_uom_qty
+            for move in self.picking_id.move_lines
+        )
+        if moves_to_process_ok:
+            self.current_filter_product = False
+            self.current_move_id = False
+            self.current_move_line_id = False
+        return moves_to_process_ok
+
+    def process_select_move(self):
+        self.next_step()
+        # Select the move line to process for the remaining qty
+        # NOTE: we should always have one move line available since we run
+        # 'action_assign' on the picking each time we validate a move.
+        move_line = fields.first(self.current_move_id.move_line_ids)
+        self.current_move_line_id = move_line
+
+    def _before_select_move_to_set_lot_number(self):
+        """Decide if we have to handle lots on the current move."""
+        return self.current_move_id.has_tracking != "none"
```

Comment:
we don't have different handlings for serial tracking and lots here?
in meaning, if the lot has tracking by serial we can constrain user with qty == 1, before sending data to ORM